Sdg23 update user guide installation instructions

namiroues opened the pull request:
This PR updates the installation instructions in the User Guide as part of the SDG documentation cleanup.

This PR is linked to MDAnalysis/mdanalysis#4927

This PR fixes #402 and #405



<!-- readthedocs-preview mdanalysisuserguide start -->
----
📚 Documentation preview 📚: https://mdanalysisuserguide--404.org.readthedocs.build/en/404/

<!-- readthedocs-preview mdanalysisuserguide end -->

namiroues committed:
Redesign installation guide in User Guide

namiroues committed:
final clear version of Installation Guide

lilyminium commented on doc/source/installation.rst:
```diff
@@ -1,130 +1,126 @@
+`pip <https://pip.pypa.io/en/latest/index.html>`_ (most common, for general users),
+`conda <https://docs.conda.io/projects/conda/en/latest/>`_ (recommended for better dependency management),
+or from `source <https://github.com/MDAnalysis/mdanalysis/>`_ (for development and contributing). Currently, 
+the `conda <https://docs.conda.io/projects/conda/en/latest/>`_ releases only support serial calculations. If you plan to use the parallel OpenMP algorithms, 
```
Since you're linking conda and pip multiple times on this page, it might be easier to [separate the text and address:](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#external-links)

```
the `conda`_
< rest of text >

.. _conda: https://docs.conda.io/projects/conda/en/latest/>
```

lilyminium replied:
Although actually it would be more informative to link to the subsections on this page using [references](https://www.sphinx-doc.org/en/master/usage/referencing.html#role-ref)!

lilyminium commented on doc/source/installation.rst:
```diff
@@ -1,130 +1,126 @@
-MDAnalysis or using the example files, you most likely don't need the tests. If you want to
-run examples in the User Guide, install the tests.
-The tests are distributed separately from the main package.
+**MDAnalysisTests** is a optional and it is a separate :ref:`test suite <mdanalysistests>` used for verifying MDAnalysis installations and running User Guide examples. 
```
Typo?

```suggestion
**MDAnalysisTests** is optional and it is a separate :ref:`test suite <mdanalysistests>` used for verifying MDAnalysis installations and running User Guide examples. 
```

lilyminium commented on doc/source/installation.rst:
```diff
@@ -1,130 +1,126 @@
     mamba update mdanalysis
 
-To install the tests:
+To install the `test suite <https://userguide.mdanalysis.org/stable/datasets.html#mdanalysistests>`_ use:
```
It might be easier to use a `:ref:` here like above, which should automatically resolve to the link!

Edit: by "above" I mean this part: ```:ref:`test suite <mdanalysistests>` ```

lilyminium reviewed (commented):
Awesome work @namiroues! I've just left some non-blocking recommendations to separate links or maybe use refs to streamline the text a bit.

namiroues committed:
update installation instructions in User Guide and address review comments

namiroues commented:
Hi @lilyminium, thanks a lot for the comments, I addressed them and I also refined the instructions to have a clearer version.

namiroues commented:
Hi @micaela-matta could you please check the changes?

orbeckst commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,150 @@
 
-conda
-=====
+conda 
```
Call it mamba/conda or Installation with conda-forge packages ... consistent with what we're doing elsewhere.

orbeckst commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,150 @@
+conda 
+======
+ 
+For most users, `conda`_ is the easiest way to install MDAnalysis because it manages dependencies effectively. 
+We highly recommend creating a new environment for MDAnalysis to ensure a clean installation of MDAnalysis and 
+its dependencies. We further recommend that you install and use `mamba`_, a faster drop-in replacement for `conda`_. 
```
Make consistent with other instructions that recommend mamba first.

namiroues replied:
Thanks @orbeckst for your comments. I was wondering if it is a good idea to recommend both options for installing mamba, via conda (if you have it installed) or via miniforge (recommended)?

orbeckst commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,150 @@
+Install from source
+===================
 
+If you plan to develop or contribute to MDAnalysis, follow these instructions first. Then, refer to :ref:`contributing` for additional guidelines.
```
Distinguish "install from source" (which really should be installing from a  release branch) vs "installing for development" (installing develop). Talk about the git branches.

"follow these instructions" is not clear — just what follows next?

Link to the specific part in contributing where we install the development version.

orbeckst commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,150 @@
 
+If you plan to develop or contribute to MDAnalysis, follow these instructions first. Then, refer to :ref:`contributing` for additional guidelines.
+
+To install the latest development versions of MDAnalysis (including unreleased features and bug fixes) you can compile it from `source`_. Use the following commands:
```
Our "from source" instructions are not very good because the **assuming you have already installed required dependencies** really shifts *all* the responsibility on the user to maintain a clean environment. 

We had discussions about shipping a conda env file but we have not achieved any kind of consensus there so the most accurate instructions are https://userguide.mdanalysis.org/stable/contributing_code.html#with-conda-forge-packages and then https://userguide.mdanalysis.org/stable/contributing_code.html#with-mamba-conda

orbeckst commented on doc/source/installation.rst:
```diff
@@ -157,22 +167,21 @@ For example, ``extra_cflags`` can be used to tune your MDAnalysis installation f
 ===================
 
-MDAnalysisData_ is an additional package with datasets that can be used in example tutorials. You can install it with ``conda`` or ``pip``:
+MDAnalysisData_ is an additional package with datasets that can be used in example tutorials. You can install it with `conda`_ or `pip`_:
```
mamba/conda

orbeckst reviewed (changes requested):
Please see comments inline, sorry to be brief.

namiroues committed:
Address PR review comments and update outdated content

namiroues commented:
Thanks @orbeckst for your comments. I've updated the installation instructions. I also updated other parts that were outdated. Please have a look. I am sure that you will have some more comments for me :).

orbeckst commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,209 @@
+daily use, but some tutorials rely on it. The package is approximately 90 MB and is not updated frequently. 
+If you plan to use tutorial examples or contribute to MDAnalysis, installing it is recommended.
+
+MDAnalysis requires Python 3.8 or later. Ensure you have an appropriate Python version installed before proceeding.
```
Currently 3.10.

Perhaps remove and link to the CHANGELOG or the PyPi page to check the latest supported version, just so that we don't forget updating in the future.

orbeckst commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,209 @@
+MDAnalysis requires Python 3.8 or later. Ensure you have an appropriate Python version installed before proceeding.
 
 .. note::
+    MDAnalysis supports **Linux**, **macOS**, and **Windows** (Python 3.8+). 
```
remove Python 3.8

orbeckst commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,209 @@
+    - If you encounter any other issues following these instructions, seek help on `GitHub Discussions (Installation)`_.    
 
-        error: Microsoft Visual C++ 14.0 is required. Get it with "Build Tools for Visual Studio": https://visualstudio.microsoft.com/downloads/
+.. warning::
```
I think this should just be a *Note* — I'd reserve warnings for issues that may lead to incorrect results or data loss ... or something like that.

Also, there's a HOLE conda-forge package that we actually install(ed) with the mdahole kit so that sentence is outdated. Just remove HOLE example.

orbeckst commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,209 @@
+`Mambaforge`_ is a minimal `conda`_ distribution that includes `mamba`_ by default, uses the `conda-forge`_ channel 
+(preferred for MDAnalysis) and avoids clutter from `Anaconda`_ base packages.
+
+To install MDAnalysis with `Mambaforge`_, first download the `mambaforge installer`_ and then run: 
```
I think here needs to be something like "follow the mambaforge installation instructions and then you should be able to run the following commands to install MDAnalysis:"

orbeckst commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,209 @@
-    mamba install -c conda-forge mdanalysis
+    mamba install -c conda-forge MDAnalysisTests
+
+For more installation options (including Windows installers), see the `official mamba installation guide`_.
```
This sentence seems out of context. If it's about mambaforge, put it there (perhaps in a footnote?)

orbeckst commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,209 @@
     pip install --upgrade MDAnalysis
 
-If you need to install a fully-featured MDAnalysis, add the ``analysis`` tag. As with ``conda``, this will not install external programs such as `HOLE`_.
+To install a fully-featured version of MDAnalysis (including optional analysis modules), use the ``analysis`` extra tag:
```
There are additional tags available (see https://github.com/MDAnalysis/mdanalysis/blob/35d9d2e3ab08e7e6741b57fe02a7215fe3b91a6c/package/pyproject.toml#L69 )
* analysis: additional dependencies for analysis modules
* extra_formats: more file formats
* parallel: include dask as a dependency for parallel analysis beyond a single machine
* doc: all dependencies to build the documentation

orbeckst commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,209 @@
     git clone https://github.com/MDAnalysis/mdanalysis
     cd mdanalysis
-    # assuming you have already installed required dependencies
+    git checkout develop  # Switch to the development branch
```
Modern git usage
```suggestion
    git switch develop    # work on the develop branch
```

orbeckst reviewed (changes requested):
Great rewrite.

I found a few things that occurred to me (and which are generally not that obvious). Thank you for updating this page, it's been high time for an update.

namiroues committed:
address PR review comments

namiroues commented:
Thanks @orbeckst for your comments. I've addressed them.

namiroues committed:
Fix pre-commit: trim trailing whitespace and ensure EOF newline"

orbeckst commented on doc/source/installation.rst:
```diff
@@ -183,9 +257,32 @@ MDAnalysisData_ is an additional package with datasets that can be used in examp
+.. _Branches page: https://github.com/MDAnalysis/mdanalysis/branches/all
+.. _CHANGELOG: https://github.com/MDAnalysis/mdanalysis/blob/develop/package/CHANGELOG
+.. _PyPI: https://pypi.org/project/MDAnalysis/
+.. _pyproject.toml file: https://github.com/MDAnalysis/mdanalysis/blob/35d9d2e3ab08e7e6741b57fe02a7215fe3b91a6c/package/pyproject.toml#L69
```
Replace link to specific commit with generic link:
```suggestion
.. _pyproject.toml: https://github.com/MDAnalysis/mdanalysis/blob/develop/package/pyproject.toml
```

orbeckst commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,215 @@
+- ``parallel``: install dask to enable parallel analysis across multiple cores or machines.
+- ``doc``: install documentation build dependencies (e.g. Sphinx, themes).
+
+To see the full list of available extras, check the `pyproject.toml file`_.
```
```suggestion
To see the full list of currently available extras, check the ``[project.optional-dependencies]`` dependencies section in the `pyproject.toml file`_.
```

orbeckst reviewed (changes requested):
Last check before approving: Have you followed the installation instructions and can you confirm that they work?

(I had minor changes but I added them myself from suggestions.)

orbeckst commented:
The pre-commit hook check is failing for me. I am just editing on GitHub so I can't easily see where the trailing whitespace is. Can you please run it locally through your pre-commit hook to fix it, @namiroues ? Thanks.

lilyminium reviewed (approved):
I haven't tested the installation instructions myself, but otherwise LGTM! Thanks for all your work @namiroues, looks excellent!

micaela-matta reviewed (approved):
I second @orbeckst's request to double check installation instructions, but other that that, LGTM! thanks @namiroues

RMeli commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,215 @@
+Currently, the :ref:`mamba-installation` installs a version of MDAnalysis that does **not** include `OpenMP`_ 
+acceleration due to limitations in precompiled conda packages. If you need `OpenMP`_-enabled features, 
+consider installing via :ref:`pip-installation` and compiling from source with a compatible compiler and 
+OpenMP support.
```
```suggestion
OpenMP support (on macOS, be aware that Apple Clang does not support OpenMP).
```
This is the most mainstream compiler not supporting OpenMP, I think.

I have some old notes about this here: https://rmeli.github.io/TIL/python/mdanalysis/#openmp-on-macos

RMeli commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,215 @@
+OpenMP support.
+
+**MDAnalysisTests** is optional and is a separate :ref:`test suite <mdanalysistests>` used 
+for verifying MDAnalysis installations and running User Guide examples. It is not required for 
```
```suggestion
for verifying MDAnalysis installations and running the User Guide examples. It is not required for 
```
?

RMeli commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,215 @@
 .. code-block:: bash
 
-    mamba install -c conda-forge mdanalysis
+    mamba install -c conda-forge MDAnalysisTests
```
```suggestion
    mamba install -c conda-forge MDAnalysisTests
```
Is the explicit channel really needed here? I don't think so. It is also not used to install MDAnalysis above, so I'd remove it to keep consistency (if it works without).

namiroues replied:
During testing, installing MDAnalysisTests without it threw an error, while MDAnalysis installed fine. I added -c conda-forge above for consistency and to avoid potential issues in other setups.

RMeli commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,215 @@
 .. code-block:: bash
 
-    mamba install -c conda-forge MDAnalysisTests
+    mamba install -c conda-forge ipywidgets
```
```suggestion
    mamba install ipywidgets
```
Same as above?

RMeli commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,215 @@
 
-If you intend to use MDAnalysis in JupyterLab, you will have to install
-an extra package for the progress bar in analysis classes:
+If you already use `conda`_, you can add `mamba`_ to your base environment:
 
 .. code-block:: bash
 
-    conda install -c conda-forge nodejs
-    jupyter labextension install @jupyter-widgets/jupyterlab-manager
+    conda install -c conda-forge mamba
```
Is this relevant? I don't find it in the current documentation, but in the old documentation this was [explicitly _not recommended_](https://web.archive.org/web/20240104003900/https://mamba.readthedocs.io/en/latest/installation/mamba-installation.html#existing-conda-install-not-recommended). If we really want to provide this, I think we need to highlight that this only works (does it?) in the base environment. It is mentioned, but I don't think it is highlighted enough, and we should probably use a warning box like in the `mamba` documentation.

orbeckst replied:
It works in base and the mamba instructions had just said to _only install in base_ #405 . In our old docs we somewhere incorrectly suggested to install into the mdanalysis venv and this PR fixes that IIRC.

(Admittedly, I install mamba into any existing base that were generated years ago with anaconda and then I remove the default channel... too much hassle to find out how to do a fresh mambaforge installation.)

orbeckst replied:
Let's just remove it all. As long as there are links to installation instructions for mamba or conda/anaconda, user will find out how to do it.

RMeli commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,215 @@
     pip install --upgrade MDAnalysis
 
-If you need to install a fully-featured MDAnalysis, add the ``analysis`` tag. As with ``conda``, this will not install external programs such as `HOLE`_.
+MDAnalysis offers several optional dependency groups (“extra tags”) that you can install with pip to enable additional features:
```
```suggestion
MDAnalysis offers several optional dependency groups (“extra tags”) that you can install with ``pip`` to enable additional features:
```
For consistency.

RMeli commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,215 @@
+
+- ``analysis``: optional dependencies for various analysis modules.
+- ``extra_formats``: support for additional file formats.
+- ``parallel``: install dask to enable parallel analysis across multiple cores or machines.
```
```suggestion
- ``parallel``: install ``dask`` to enable parallel analysis across multiple cores or machines.
```
Maybe it is worth adding a link to [dask](https://www.dask.org)?

RMeli commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,215 @@
+
+To see the full list of currently available extras, check the ``[project.optional-dependencies]`` dependencies section in the `pyproject.toml file`_.
+
+To install with optional features, use pip with one or more extras:
```
```suggestion
To install with optional features, use ``pip`` with one or more extras:
```

RMeli commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,215 @@
+    mamba activate mdanalysis
 
-To install the latest stable version of MDAnalysis via ``conda``, use the following command. This installs all dependencies needed for full analysis functionality (excluding external programs such as `HOLE`_):
+To install the :ref:`test suite <mdanalysistests>` (optional, ~90 MB):
```
```suggestion
To install the :ref:`test suite <mdanalysistests>` (optional):
```
This is the kind of information that is bound to get out of date. Since 90MB is essentially nothing in today's standards, it might be worth simply dropping it IMO.

RMeli commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,215 @@
+    pip install --upgrade MDAnalysis[analysis,extra_formats,parallel]
 
-To install/upgrade tests:
+To install or upgrade the :ref:`test suite <mdanalysistests>` (optional, ~90 MB):
```
```suggestion
To install or upgrade the :ref:`test suite <mdanalysistests>` (optional):
```
See above.

RMeli commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,215 @@
-Development versions
-====================
-To install development versions of MDAnalysis, you can compile it from source. In order to install from source, you will need ``numpy`` and ``cython``. See :ref:`create-virtual-environment` for instructions on how to create a full development environment.
+Install from source
+===================
```
Is this as a separate section really necessary? Can't we just have the developer installation (and eventually mention not to use the editable `-e` option)? I would unify the two sections, but this might just be personal preference so feel free to ignore if no-one else feels the same.

orbeckst replied:
I would keep them separate. The "Install from source" has information about the package branches. 

The "Develop installation" only uses the develop branch. 

Btw, what's so bad about `pip -e` ?

RMeli replied:
> and eventually mention not to use the editable -e option

Sorry, I was unclear. There is nothing wrong with -e. What I was thinking was that we could have a single set of instructions and mention the different branches, and that if one is not interested in developing (i.e. editing the source), the -e option is not necessary.

But I'm OK to keep the sections separate, I just thought there was a lot of overlap.

RMeli commented on doc/source/installation.rst:
```diff
@@ -148,7 +223,7 @@ You can pass any additional compiler flags for the C/C++ compiler using the ``ex
 
 For example, ``extra_cflags`` can be used to tune your MDAnalysis installation for your current architecture using the `-march`, `-mtune`, `-mcpu` and related compiler flags.
-*Which* particular compiler flags to use depends on your CPU architecture. An example for an x86_64 machine would be to change the line in `setup.cfg` as follows:
+The optimal compiler flags depend on your CPU architecture. An example for an x86_64 machine would be to change the line in `setup.cfg` as follows:
```
I don't think this suggestion should be limited to `x86_64`. ` -march=native -mtune=native` should work for different architectures. Maybe it was intended to use something like `-march=x86-64-v4`? Is there a better example we can use?

RMeli commented on doc/source/installation.rst:
```diff
@@ -157,22 +232,21 @@ For example, ``extra_cflags`` can be used to tune your MDAnalysis installation f
 
-Be sure to use the recommended flags for your target architecture. For example, ARM platforms recommend using ``-mcpu`` *instead* of ``-mcpu``, while
-PowerPC platforms prefer *both* ``-mcpu`` and ``-mtune``.
+Be sure to use the recommended flags for your target architecture. For example, ARM platforms recommend using ``-mcpu`` instead of ``-march``, while 
```
Is this still the case? GCC documentation now says the following about `-march=name`:

> This option can be used in conjunction with or instead of the -mcpu= option.

And under `-mcpu`:

> Where this option is used in conjunction with `-march` or `-mtune`, those options take precedence over the appropriate part of this option.

I'd double check if this information is still relevant. I'd point directly to the compiler documentation here, because what we write here might get out of date at some point.

RMeli reviewed (commented):
Thanks @namiroues, looks very good! I left some additional comments.

orbeckst commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,215 @@
-    mamba install -c conda-forge MDAnalysisTests
+    mamba install -c conda-forge ipywidgets
+
+**2. Alternative: Install mamba to an existing conda installation**
```
Following the removal of the "install mamba into existing conda env" from https://mamba.readthedocs.io/en/latest/installation/mamba-installation.html#existing-conda-install-not-recommended , I would **remove this whole section** and just link to the instructions for either mamba or anaconda.

orbeckst replied:
This could become an "Installation with conda from conda-forge".

Then all the `-c conda-forge` become relevant.

orbeckst commented:
The RTD build failed because a package had the wrong checksum. I assume that this is an intermittent error (which is out of our control) and it will work the next time you push to this PR.

namiroues committed:
address PR review comments

namiroues commented:
Thanks @orbeckst and @RMeli for your comments. I addressed the comments and tested the installation instructions. They work for me with conda-forge (so I explicitly added this). Can someone try a fresh installation with mambaforge to double-test this?

namiroues committed:
fix missing newline at end of installation.rst

orbeckst commented on doc/source/installation.rst:
```diff
@@ -183,9 +260,35 @@ MDAnalysisData_ is an additional package with datasets that can be used in examp
+.. _Branches page: https://github.com/MDAnalysis/mdanalysis/branches/all
+.. _CHANGELOG: https://github.com/MDAnalysis/mdanalysis/blob/develop/package/CHANGELOG
+.. _PyPI: https://pypi.org/project/MDAnalysis/
+.. _pyproject.toml file: https://github.com/MDAnalysis/mdanalysis/blob/35d9d2e3ab08e7e6741b57fe02a7215fe3b91a6c/package/pyproject.toml#L69
```
Link to the generic latest file, not a specific commit. Otherwise the docs are stuck on this specific version.

orbeckst replied:
```suggestion
.. _pyproject.toml file: https://github.com/MDAnalysis/mdanalysis/blob/develop/package/pyproject.toml#L69
```

orbeckst commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,215 @@
+
+    git clone https://github.com/MDAnalysis/mdanalysis
+    cd mdanalysis
+    git checkout release-<latest-version>
```
Is `git switch` preferred over `git checkout` nowadays?

According to https://stackoverflow.com/a/70454786 simple checkout and switch are equivalent (but switch is clearer):
```suggestion
    git switch release-<latest-version>
```

orbeckst reviewed (changes requested):
Minor changes inline, the instructions themselves seem good, but not everything works. I kept notes for future follow-up.

# Notes from testing on macOS x86_64

## ❌  mamba Python 3.13.2

I tried in an existing conda/mamba installation. By default it created a Python 3.13.2 environment. MDAnalysis installed fine:
```
>>> python -c "import MDAnalysis; print(MDAnalysis.__version__)"
2.9.0
```

However, MDAnalysisTests has issues for me:
```
mamba install -c conda-forge MDAnalysisTests
```
gives
```
Getting  conda-forge osx-64
Getting  conda-forge noarch

Looking for: ['mdanalysistests', 'python 3.13.2']

509175 packages in https://conda.anaconda.org/conda-forge/osx-64
259522 packages in https://conda.anaconda.org/conda-forge/noarch
Encountered problems while solving.
Problem: nothing provides mdanalysis 0.15.0 needed by mdanalysistests-0.15.0-py27_0
```
I have no idea why it wants to install 0.15.

Then I try pinning
```
mamba install -c conda-forge "MDAnalysisTests=2.9.0"
```
and that fails with
```
Looking for: ['mdanalysistests 2.9.0.*', 'python 3.13.2']

509175 packages in https://conda.anaconda.org/conda-forge/osx-64
259522 packages in https://conda.anaconda.org/conda-forge/noarch
Encountered problems while solving.
Problem: package mdanalysistests-2.9.0-pyhd8ed1ab_0 requires python >=3.9,<=3.13, but none of the providers can be installed
```
The mdanalysistests condaforge recipe contains https://github.com/conda-forge/mdanalysistests-feedstock/blob/c9c9a496f8172b2bc9a5c7571092e2ab2f1f8c2f/recipe/meta.yaml#L25
```yaml
    - python >=3.9,<=3.13
```
Based on the [package match specification](https://docs.conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html#package-match-specifications) this seems to be the issue because it will **not** match Python 3.13.2.

I raised https://github.com/conda-forge/mdanalysistests-feedstock/issues/51, which has already been fixed. (So need to test with conda -forge packages ...)

## ✅ mamba Python 3.12
```
mamba create -n mdanalysis -c conda-forge mdanalysis python=3.12
mamba activate mdanalysis
mamba install -c conda-forge MDAnalysisTests
```
Tests in parallel 
```
pytest --disable-pytest-warnings -n 2 -v --pyargs MDAnalysisTests
```
were a bit wonky for me. Deadlocked once around 87%, second time it happened I started killing python3.12 processes and then suddenly tests continued to run (although possibly only in one runner). Tried again
```
pytest --disable-pytest-warnings -n 4 -v --pyargs MDAnalysisTests
``` 
succeeded (installs 2.9.0 and tests pass).
```
============ 19714 passed, 363 skipped, 7 xfailed, 2 xpassed, 107840 warnings in 749.94s (0:12:29) ============
```

## ✅ , ❌ pip 3.13
Create mamba env with python 3.13 
```
mamba create -n mdapip313 python=3.13 pip
```
### ✅ standard
and installed with
```
pip install --upgrade MDAnalysis
pip install --upgrade MDAnalysisTests
```
Tests 
```
pytest --disable-pytest-warnings -n 2 -v --pyargs MDAnalysisTests
```
passed
```
============ 19014 passed, 781 skipped, 6 xfailed, 2 xpassed, 89945 warnings in 962.55s (0:16:02) =============
```

### ❌ Installing extras
```
pip install --upgrade MDAnalysis[analysis,extra_formats,parallel]
```
failed because pytng did not build
```
...
      pytng/pytng.c:2398:22: error: implicit declaration of function '_PyList_Extend' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
          PyObject* none = _PyList_Extend((PyListObject*)L, v);
                           ^
...
      pytng/pytng.c:2436:39: error: implicit declaration of function '_PyInterpreterState_GetConfig' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
          __pyx_assertions_enabled_flag = ! _PyInterpreterState_GetConfig(__Pyx_PyThreadState_Current->interp)->optimization_level;
                                            ^
      pytng/pytng.c:2436:107: error: member reference type 'int' is not a pointer
          __pyx_assertions_enabled_flag = ! _PyInterpreterState_GetConfig(__Pyx_PyThreadState_Current->interp)->optimization_level;
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~  ^
...
      pytng/pytng.c:34856:13: error: implicit declaration of function '_PyUnicode_FastCopyCharacters' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                  _PyUnicode_FastCopyCharacters(result_uval, char_pos, uval, 0, ulength);
                  ^
...
      pytng/pytng.c:38679:70: error: too few arguments to function call, expected 6, have 5
                                                    is_little, !is_unsigned);
                                                                           ^
...
      pytng/pytng.c:34856:13: error: implicit declaration of function '_PyUnicode_FastCopyCharacters' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                  _PyUnicode_FastCopyCharacters(result_uval, char_pos, uval, 0, ulength);
                  ^
...
      pytng/pytng.c:38679:70: error: too few arguments to function call, expected 6, have 5
                                                    is_little, !is_unsigned);
                                                                           ^
...
              #define Py_EXPORTED_SYMBOL __attribute__ ((visibility ("default")))
                                         ^
      pytng/pytng.c:39895:13: error: implicit declaration of function '_PyGen_SetStopIterationValue' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                  _PyGen_SetStopIterationValue(result);
                  ^
...
      22 warnings and 9 errors generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]

...

Failed to build pytng
ERROR: Failed to build installable wheels for some pyproject.toml based projects (pytng)
```

orbeckst committed:
Update doc/source/installation.rst

orbeckst committed:
Update doc/source/installation.rst

orbeckst commented on doc/source/installation.rst:
```diff
@@ -1,140 +1,215 @@
+
+    git clone https://github.com/MDAnalysis/mdanalysis
+    cd mdanalysis
+    git switch release-<latest-version>
```
Does not work for me on a fresh clone – neither using "package-..." nor "release-..."
```
git switch release-2.9.0
fatal: a branch is expected, got tag 'release-2.9.0'
hint: If you want to detach HEAD at the commit, try again with the --detach option.
```
or branch name
```
$ git switch package-2.9.0
fatal: a branch is expected, got tag 'package-2.9.0'
hint: If you want to detach HEAD at the commit, try again with the --detach option.
```

orbeckst replied:
Using `git checkout`
```
 git checkout package-2.9.0
Note: switching to 'package-2.9.0'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false
```

orbeckst reviewed (changes requested):
Source installation instructions do not work as written.

Please fix/clarify.

namiroues committed:
address PR review comments

namiroues commented:
Thanks @orbeckst. I fixed it and tested it myself.

orbeckst reviewed (approved):
Good rewrite, all perfect from my end.

orbeckst commented:
Nice work, @namiroues !
